### ISSUE TITLE
Training produces broken deltas when another delta is loaded

### ISSUE THREAD

**Issue by doomed151** [guaneec/custom-diffusion-webui#2]
Steps to reproduce:
1. Go to txt2img tab
2. Enter something in the prompt
3. Select an existing delta from the "Tuned weights" dropdown
4. Click "Generate"
5. Go to "Train" tab
6. Create a new embedding
7. Go to "Train Custom Diffusion"
8. Fill in your training parameters and click "Train Embedding"
9. Once training is done, go back to txt2img tab
10. Modify the prompt to include your embedding name
11. Choose the new delta from the "Tuned weights" dropdown
12. Click "Generate"
13. Get broken image

It seems the script doesn't restore the original weights when training. I might be wrong since I'm not very familiar with the code yet. Will open a PR if I'm successful in fixing it.

**Comment by guaneec** [comment]
Thought nobody would notice this :P
Will fix within about 1 day

**Comment by guaneec** [comment]
Just pushed the fix to the dev branch.
Will merge into master when I have the time to properly test it and clean it up.
For the meantime. just remember to manually unload the deltas before training.

**Comment by doomed151** [comment]
Appreciate the quick fix! I'll be sure to check it out.
